sources/packages/api/src/services/designation-agreement/designation-agreement.service.ts

```
@@ -147,4 +163,166 @@ export class DesignationAgreementService extends RecordDataModelService<Designat
+    searchCriteria?: string,
+  ): Promise<DesignationAgreement[]> {
+    const designationQuery = this.repo
+      .createQueryBuilder("designation")
+      .select([
+        "designation.id",
+        "designation.designationStatus",
+        "designation.submittedDate",
+        "designation.startDate",
+        "designation.endDate",
+        "institution.legalOperatingName",
+      ])
+      .innerJoin("designation.institution", "institution")
+      .where(`${filterColumn} = :filterValue`, { filterValue });
+    if (searchCriteria) {
+      designationQuery.andWhere(
+        "institution.legalOperatingName Ilike :searchCriteria",
+        { searchCriteria: `%${searchCriteria}%` },
+      );
+    }
+    return designationQuery
+      .orderBy("designation.designationStatus", "ASC")
+      .addOrderBy("designation.submittedDate", "DESC")
+      .getMany();
+  }
+
+  /**
+   * Private method to build designation locations for update.
+   * This builder is only for designation Approval.
+   * @param locationsDesignations
+   * @param designationLocations
+   * @param userId
+   * @returns Designation locations for approval.
+   */
+  private buildDesignationLocations(
+    locationsDesignations: UpdateDesignationLocation[],
+    designationLocations: DesignationAgreementLocation[],
+    userId: number,
+  ): DesignationAgreementLocation[] {
+    return locationsDesignations.map(
+      (locationPayload: UpdateDesignationLocation) => {
+        const location = new DesignationAgreementLocation();
+
+        location.approved = locationPayload.approved;
+        location.institutionLocation = {
+          id: locationPayload.locationId,
+        } as InstitutionLocation;
+        location.creator = { id: userId } as User;
+        location.requested = false;
+        const designationLocation = designationLocations.find(
+          (item) => item.institutionLocation.id === locationPayload.locationId,
+        );
+        if (designationLocation) {
```

I believe that if the _designationLocation_ is not defined at this point we can just throw an exception.
Returning the object without the _locationId_ will also throw an exception later in the process but maybe it will be not so clear.

---

If the designationLocation is not defined, then it is treated as insert here.